src/common/arrow/arrow_converter.cpp

```diff
@@ -127,8 +130,24 @@ void SetArrowFormat(DuckDBArrowSchemaHolder &root_holder, ArrowSchema &child, co
 	case LogicalTypeId::DOUBLE:
 		child.format = "g";
 		break;
-	case LogicalTypeId::UUID:
+	case LogicalTypeId::UUID: {
+		// This is a canonical extension, hence needs the "arrow." prefix
+		child.format = "w:16";
+		auto schema_metadata = ArrowSchemaMetadata();
+		schema_metadata.AddOption(ArrowSchemaMetadata::ARROW_EXTENSION_NAME, "arrow.uuid");
+		schema_metadata.AddOption(ArrowSchemaMetadata::ARROW_METADATA_KEY, "");
+		root_holder.metadata_info.emplace_back(schema_metadata.SerializeMetadata());
+		child.metadata = root_holder.metadata_info.back().get();
+		break;
+	}
 	case LogicalTypeId::VARCHAR:
+		if (type.IsJSONType()) {
+			auto schema_metadata = ArrowSchemaMetadata();
+			schema_metadata.AddOption(ArrowSchemaMetadata::ARROW_EXTENSION_NAME, "arrow.json");
+			schema_metadata.AddOption(ArrowSchemaMetadata::ARROW_METADATA_KEY, "");
+			root_holder.metadata_info.emplace_back(schema_metadata.SerializeMetadata());
+			child.metadata = root_holder.metadata_info.back().get();
+		}
 		if (options.produce_arrow_string_view) {
 			child.format = "vu";
```

Comment:
Just a question mostly, does the `arrow.json` extension type not expect a specific underlying logical type? Where perhaps `string_view` is not supported for example?

Reply:
From https://github.com/duckdblabs/voltron-data/issues/12 it's explained that `string_view`s are fair game